src/CarbonAware.WebApi/src/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -7,13 +9,25 @@ public static void AddMonitoringAndTelemetry(this IServiceCollection services, I
         
         var envVars = configuration?.GetSection(CarbonAwareVariablesConfiguration.Key).Get<CarbonAwareVariablesConfiguration>();
         var telemetryProvider = GetTelemetryProviderFromValue(envVars?.TelemetryProvider);
-
+        var logger = CreateConsoleLogger(configuration);
         switch (telemetryProvider) {
             case TelemetryProviderType.ApplicationInsights:
             {
-                if (!String.IsNullOrEmpty(configuration?["ApplicationInsights_Connection_String"]))
+                string? instrumentationKey = configuration?["AppInsights_InstrumentationKey"];
+
+                if (!String.IsNullOrEmpty(configuration?["ApplicationInsights_Connection_String"])) 
                 {
+                    logger.LogInformation("Application Insights connection string found");
                     services.AddApplicationInsightsTelemetry();
+                } 
+                else if (!String.IsNullOrEmpty(instrumentationKey)) 
+                {
+                    logger.LogInformation("Application Insights Instrumentation Key found");
+                    services.AddApplicationInsightsTelemetry(instrumentationKey);
```

Comment:
```suggestion
                    services.AddApplicationInsightsTelemetry();
```

This works the same if the env `AppInsights_InstrumentationKey` is set. So no need to pass the argument.

Reply:
When @Kanishk and I tested this, it wasnt working without the InstrumentationKey passed in.

Reply:
> When @kanishk and I tested this, it wasnt working without the InstrumentationKey passed in.

I have tried this way and this is how also the customer tried. As long the environment variable is `AppInsights_InstrumentationKey` and the value is the correct one, it should work.